File: src/main/java/com/baulsupp/oksocial/services/lyft/LyftUtil.java
package com.baulsupp.oksocial.services.lyft;

import com.google.common.collect.Sets;
import java.util.Arrays;
import java.util.Collection;
import java.util.Collections;
import java.util.Set;
import okhttp3.Request;

public class LyftUtil {
  public static final Collection<String> SCOPES = Arrays.asList("public",
      "rides.read",
      "offline",
      "rides.request",
      "profile");

  private LyftUtil() {
  }

  public static final Set<String> API_HOSTS =
      Collections.unmodifiableSet(Sets.newHashSet(
          "api.lyft.com")
      );

  public static Request apiRequest(String s, Request.Builder requestBuilder) {
    return requestBuilder.url("https://api.lyft.com" + s).build();
  }
}

File: src/main/kotlin/com/baulsupp/oksocial/services/lyft/LyftUtil.kt
package com.baulsupp.oksocial.services.lyft

import okhttp3.Request
import java.util.Arrays

object LyftUtil {
  val SCOPES: Collection<String> = Arrays.asList("public",
      "rides.read",
      "offline",
      "rides.request",
      "profile")

  val API_HOSTS = setOf((
      "api.lyft.com")
  )

  fun apiRequest(s: String, requestBuilder: Request.Builder): Request {
    return requestBuilder.url("https://api.lyft.com" + s).build()
  }
}
